source/actions3/BioXAS/BioXASMAXvMotorMoveToLimitAction.cpp

```diff
@@ -0,0 +1,129 @@
+#include "BioXASMAXvMotorMoveToLimitAction.h"
+#include "beamline/AMBeamline.h"
+#include "util/AMErrorMonitor.h"
+
+BioXASMAXvMotorMoveToLimitAction::BioXASMAXvMotorMoveToLimitAction(BioXASMAXvMotorMoveToLimitActionInfo *info, BioXASMAXvMotor *motor, QObject *parent) :
+	AMAction3(info, parent)
+{
+	control_ = motor;
+	initialValue_ = 0;
+}
+
+BioXASMAXvMotorMoveToLimitAction::BioXASMAXvMotorMoveToLimitAction(const BioXASMAXvMotorMoveToLimitAction &original) :
+	AMAction3(original)
+{
+	control_ = original.control();
+	initialValue_ = 0;
+}
+
+BioXASMAXvMotorMoveToLimitAction::~BioXASMAXvMotorMoveToLimitAction()
+{
+
+}
+
+void BioXASMAXvMotorMoveToLimitAction::onMotorMoveStarted()
+{
+	// Set the action as started.
+
+	setStarted();
+
+	// Start progress timer updates.
+
+	connect( &progressTimer_, SIGNAL(timeout()), this, SLOT(onProgressTick()) );
+	progressTimer_.start(250);
+}
+
+void BioXASMAXvMotorMoveToLimitAction::onMotorMoveFailed()
+{
+	// Disconnect from control.
+
+	disconnect( control_, 0, this, 0 );
+
+	// End progress timer updates.
+
+	progressTimer_.stop();
+	disconnect( &progressTimer_, 0, this, 0 );
+
+	// Check whether the control made it to the appropriate limit.
+	// If so, the action has succeeded. If not, the action has failed.
+
+	if (atLimit()) {
+		setProgress(100, 100);
+		setSucceeded();
+
+	} else {
+		setFailed(QString("Failed to move motor '%1' to '%2' limit. The move failed without reaching the limit.").arg(control_->name()).arg(moveToLimitInfo()->limitSetpointToString(moveToLimitInfo()->limitSetpoint())));
+	}
+}
+
+void BioXASMAXvMotorMoveToLimitAction::onMotorMoveSucceeded()
```

Comment:
If this and the above onMotorMoveFailed() contain the exact same code could you not hook both signals up to one slot?
onMotorMoveFinished()?